src/kOS/Suffixed/AggregateResourceValue.cs

```diff
@@ -1,44 +1,41 @@
 ﻿using System.Collections.Generic;
 using kOS.Safe.Encapsulation;
+using kOS.Safe.Encapsulation.Suffixes;
+using kOS.Suffixed.Part;
 
 namespace kOS.Suffixed
 {
     public class AggregateResourceValue : Structure
     {
         private readonly string name;
+        private readonly SharedObjects shared;
         private double amount;
         private double capacity;
-        private readonly ListValue parts;
+        private readonly SuffixedList<PartValue> parts;
 
-        public AggregateResourceValue(string name)
+        public AggregateResourceValue(string name, SharedObjects shared)
         {
             this.name = name;
+            this.shared = shared;
             amount = 0;
             capacity = 0;
-            parts = new ListValue();
+            parts = new SuffixedList<PartValue>();
+            InitializeAggregateResourceSuffixes();
         }
 
-        public override object GetSuffix(string suffixName)
+        private void InitializeAggregateResourceSuffixes()
         {
-            switch (suffixName)
-            {
-                case "NAME":
-                    return name;
-                case "AMOUNT":
-                    return amount;
-                case "CAPACITY":
-                    return capacity;
-                case "PARTS":
-                    return parts;
-            }
-            return base.GetSuffix(suffixName);
+            AddSuffix("NAME", new Suffix<string>(() => name));
+            AddSuffix("AMOUNT", new Suffix<double>(() => amount));
+            AddSuffix("CAPICITY", new Suffix<double>(() => capacity));
+            AddSuffix("PARTS", new Suffix<SuffixedList<PartValue>>(() => parts));
```

Comment:
before this change we were returning a list of ksp parts that would be useless. Because we have so many list suffixes i think i will make a new suffix type to make

    Suffix<SuffixedList<PartValue>>

into
    ListSuffix<PartValue>